docs/topics/auth/passwords.txt

```diff
@@ -474,14 +474,18 @@ The :mod:`django.contrib.auth.hashers` module provides a set of functions
 to create and validate hashed passwords. You can use them independently
 from the ``User`` model.
 
-.. function:: check_password(password, encoded)
+.. function:: check_password(password, encoded, setter=None, preferred="default")
 
     If you'd like to manually authenticate a user by comparing a plain-text
     password to the hashed password in the database, use the convenience
-    function :func:`check_password`. It takes two arguments: the plain-text
-    password to check, and the full value of a user's ``password`` field in the
-    database to check against, and returns ``True`` if they match, ``False``
-    otherwise.
+    function :func:`check_password`. It takes mandatory two arguments: the
+    plain-text password to check, and the full value of a user's ``password``
+    field in the database to check against, and returns ``True`` if they match,
+    ``False`` otherwise. Optionally, you can pass a callable ``setter`` that
+    takes the password and will be called when you need to regenerate it. You
+    can also pass ``preferred`` to change a hashing algorithm, if you don't
+    want to use the default (first entry of ``PASSWORD_HASHERS`` setting). See
+    :ref:`auth-included-hashers` for the algorithm name of each hasher.
```

Comment:
```suggestion
    function :func:`check_password`. It takes two mandatory arguments: the
    plain-text password to check and the full value of a user's ``password``
    field in the database to check against. It returns ``True`` if they match,
    ``False`` otherwise. Optionally you can pass a callable ``setter`` that
    takes the password and will be called when you need to regenerate it. You
    can also pass ``preferred`` to change a hashing algorithm if you don't
    want to use the default (first entry of ``PASSWORD_HASHERS`` setting). See
    :ref:`auth-included-hashers` for the algorithm name of each hasher.
```

A few suggestions/corrections:

- Just needed to put the quantifier ("two") before the "mandatory"
- The <something>, and <something>, and <something> was a little hard to read (also I'm pretty sure the comma for the first "and" was grammatically incorrect)
- (Also a lot of commas break the paragraph's flow)

Reply:
@shangxiao Thanks, pushed edits.